### PR TITLE
libhio: add libhio package to spack

### PR DESCRIPTION
**Pull request by hppritcha**
spack brought out some configury/make issues with
libhio, so some patches come along for the ride.

Signed-off-by: Howard Pritchard <howardp@lanl.gov>

### CODE REVIEW

**Comment by hppritcha** [comment]
I'm failing to grock the codecov/project, etc. failures.  What do I need to do to reverse the downward coverage?

File: var/spack/repos/builtin/packages/libhio/package.py [hunk 1]
```diff
@@ -22,20 +22,58 @@
...
+
+    #
+    # wow, we need to patch libhio
+    #
+    patch('0001-configury-fix-a-problem-with-bz2-configury.patch', when="@1.4.1.0")
+    patch('0001-hdf5-make-docs-optional.patch', when="@1.4.1.0")
+
+    def autoreconf(self, spec, prefix):
+            autoreconf = which('autoreconf')
+            autoreconf('-ifv')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libhio/package.py]: These lines are indented too far.

File: var/spack/repos/builtin/packages/libhio/package.py [hunk 2]
```diff
@@ -22,20 +22,58 @@
...
+
+    def configure_args(self):
+        spec = self.spec
+        args = []
+
+        #
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libhio/package.py]: Probably don't need this comment.

File: var/spack/repos/builtin/packages/libhio/package.py [hunk 3]
```diff
@@ -22,20 +22,58 @@
...
+    # We don't include older versions since they are missing features
+    # needed by current and future consumers of libhio
+    #
+    version('1.4.1.0', '6ef566fd8cf31fdcd05fab01dd3fae44')
+
+    #
+    # main users of libhio thru spack will want to use HFDF5 plugin,
+    # so make hdf5 variant a default
+    #
+    variant('hdf5',    default=True, description='Enable HDF5 support')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libhio/package.py]: No need for the extra spaces between `'hdf',` and `default=True`.

**Comment by hppritcha** [comment]
@adamjstewart sorry for the delay, fixed up the white spaces per your request.

**Pull request merged**